agent/functional_tests/tests/functionaltests_unix_test.go

```diff
@@ -1456,3 +1456,19 @@ func TestServerEndpointValidator(t *testing.T) {
 	assert.True(t, ok, "Get exit code failed")
 	assert.Equal(t, 42, code, "Wrong exit code")
 }
+
+// TestAppMeshCNIPlugin validates the functionality of appmesh cni plugin.
+func TestAppMeshCNIPlugin(t *testing.T) {
```

Comment:
You could also simplify here through test [generator](https://github.com/aws/amazon-ecs-agent/blob/master/agent/functional_tests/tests/generated/generate_tests.go) to generate this test in this [file](https://github.com/aws/amazon-ecs-agent/blob/master/agent/functional_tests/tests/generated/simpletests_unix/simpletests_generated_unix_test.go)

Reply:
Good point. Considering I will need to set up a awslog to store container log, this does not apply to my test case anymore.

Reply:
Can you elaborate why do we need awslog here instead of troubleshooting like other tests through the functional output?

Reply:
So the functional output does not give us the insight of what happened within the container. So in order to know what happened in the container based on the container log, I need to upload the container log to CW.